app/controllers/api/applications_controller.rb

```diff
@@ -1,74 +1,51 @@
 # frozen_string_literal: true
 
-class Api::ApplicationsController < Api::BaseController
-  before_action :authorize_partners
+class Api::ApplicationsController < FrontendController
+  include ThreeScale::Search::Helpers
+  include ApplicationsControllerMethods
 
+  before_action :authorize_partners
+  before_action :find_plans
   before_action :find_service
-  before_action :find_cinstance, only: %i[show edit]
+  before_action :find_buyer, only: :create
+  before_action :authorize_multiple_applications, only: :create
+  before_action :find_application_plan, only: :create
+  before_action :find_service_plan, only: :create
 
-  include ThreeScale::Search::Helpers
-  include DisplayViewPortion
-  helper DisplayViewPortion::Helper
+  helper_method :accessible_services
 
   activate_menu :serviceadmin, :applications, :listing
-  sublayout 'api/service'
-
-  def index
-    # TODO: This code is REALLY bad but it is copied and pasted from Buyers::ApplicationsController#index because
-    # doing it well requires time and we don't have time right now.
-    # Editing this action may require touching the other one
 
-    @states = Cinstance.allowed_states.collect(&:to_s).sort
-    @search = ThreeScale::Search.new(params[:search] || params)
-    service_application_plans = @service.application_plans
-    @application_plans = service_application_plans.stock
-    @stock_and_custom_application_plans = service_application_plans.size
+  sublayout 'api/service'
 
-    @search.service_id = @service.id
+  def create
+    super
 
-    if params[:application_plan_id]
-      @plan = @service.application_plans.find params[:application_plan_id]
-      @search.plan_id = @plan.id
+    if @cinstance.save
+      flash[:notice] = 'Application was successfully created.'
+      redirect_to provider_admin_application_path(@cinstance)
```

Comment:
I think it should redirect to the index page in the same scope (API product, in this case).

Reply:
Index, not show page?